test/pre-post-extension.js

```diff
@@ -0,0 +1,56 @@
+import chai, {expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import Stipulate from '../src';
+
+chai.use(chaiAsPromised);
+
+class ExtStip extends Stipulate {
```

Comment:
Ahhhh, I think I see what you're saying now. These tests are showing that 1) `beforeRequest` is being called before the request is being sent, and 2) `afterResponse` is being called after receiving the response. However, I've implemented the tests by extending the class, and (if I gather correctly) you believe the wording of the tests as well as the title of the file obfuscates what's actually being tested. If that's the case, I agree. But I do think these are valid tests, implemented in an informative way(i.e. you can look at them and see an example of how one might extend the class to use those hooks). How about just `hook-methods.js`, and rewrite the test descriptions to be more agnostic about how the hooks are implemented?